linux/README.md

```diff
@@ -39,13 +39,53 @@ Where to put caches and temporary files?
 
 ## Runtime situation
 
-Explain the complexity about Linux deployment.
-How to link against an old glibc.
-What to link statically and what to link dynamically.
+Linux comes in various different distributions, with differing sets of installed packages.
+
+The widest compatibility can usually be achieved by picking an old Debian or Ubuntu
+stable release as build platform and statically linking against library dependencies.
+Static linking increases binary size but ensures that all needed dependencies are
+shipped along/inside the plugin.
+
+A small set of libraries can create problems when statically linked and need
+to be excluded so plugins either rely on it already being present on the host system
+or make their dependencies optional (e.g. by using `dlopen(3)`).
+The [AppImage](https://github.com/AppImage) project maintains a list of problematic libraries here:
+https://github.com/AppImage/pkg2appimage/blob/master/excludelist
```

Comment:
They recommend to link dynamically to the C++ library and I'm not sure it is a good idea.

Reply:
You should dynamically link against `libstdc++` _unless_ you're using a GCC version that's newer than what's in the repos for the distro you're compiling on. In that case you should statically link it. So as an example, Ubuntu 20.04 defaults to GCC 9, but GCC 10 is available from the repos, and it also ships with the `libstdc++` from GCC 10 by default. So if you use GCC 10 on Ubuntu 20.04, then you won't need to statically link `libstdc++`. But if, say, you wanted to use GCC 10 or newer on Ubuntu 18.04, then you would need to install that first through a third party repo/PPA. In that case you _will_ need to statically link `libstdc++` because the version that comes with Ubuntu 18.04 is older than the one you used to compile your plugin against.

Reply:
This is an *excludelist*, i.e. libstdc++.so should generally *not* be included (it can be expected everywhere and maintains ABI fairly well).

The list is well maintained btw, every library issue in any AppImage project ultimately results in this list being updated.